assets/source/setup-guide/app/components/SaveSettingsButton.test.js

```diff
@@ -0,0 +1,42 @@
+jest.mock( '../helpers/effects', () => {
+	return {
+		useCreateNotice: () => () => {},
+		useSettingsSelect: () => {},
+		useSettingsDispatch: () => () => {},
+	};
+} );
+jest.mock( '@woocommerce/tracks', () => {
+	return {
+		recordEvent: jest.fn(),
+	};
+} );
+
+/**
+ * External dependencies
+ */
+import { recordEvent } from '@woocommerce/tracks';
+import { fireEvent, render } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import SaveSettingsButton from './SaveSettingsButton';
+
+afterEach( () => {
+	jest.clearAllMocks();
+} );
+
+describe( 'Save Settings function', () => {
+	const eventName = 'pfw_save_changes_button_click';
+
+	it( `${ eventName } is called on Save settings`, () => {
+		const { getByRole } = render( <SaveSettingsButton /> );
+		const saveSettingsBtn = getByRole( 'button' );
+
+		fireEvent.click( saveSettingsBtn );
+		expect( recordEvent.mock.calls[ 0 ][ 0 ] ).toBe( eventName );
+		expect( recordEvent.mock.calls[ 0 ][ 1 ].context ).toBe(
+			'pinterest_settings'
+		);
```

Comment:
Personally, I prefer custom assertions like, `.toBeCalledOnceWithExactly` as they are easier to read in code and in failure message, plus cater for things like ``Cannot read property 0 of undefined``

Reply:
Fixed in https://github.com/woocommerce/pinterest-for-woocommerce/pull/295/commits/0afbde0966e4b9124d2ae8a573d788ea88213746